onnxruntime/core/providers/cpu/tensor/grid_sample.h

```diff
@@ -15,20 +15,35 @@ template <typename T>
 class GridSample final : public OpKernel {
  public:
   explicit GridSample(const OpKernelInfo& info) : OpKernel(info) {
-    std::string mode_str = info.GetAttrOrDefault<std::string>("mode", "bilinear");
+    int start_version = info.node().SinceVersion();
+    if (start_version >= 20) {
+      std::string mode_str = info.GetAttrOrDefault<std::string>("mode", "linear");
+      ORT_ENFORCE(mode_str == "linear" || mode_str == "nearest" || mode_str == "cubic",
+                  "mode \"", mode_str, "\" not supported, expect linear, nearest or cubic");
+      if (mode_str == "cubic") {
+        mode_ = Cubic;
+      } else if (mode_str == "nearest") {
+        mode_ = Nearest;
+      } else {
```

Comment:
If would write `else if (mode_str == "linear") ... else ORT_THROW` to avoid comparing twice the strings.

Reply:
updated